src/sharding/Shard.js

```diff
@@ -96,7 +96,7 @@ class Shard extends EventEmitter {
      * @type {Function}
      * @private
      */
-    this._exitListener = this._handleExit.bind(this, undefined);
```

Comment:
This may cause issues if reverted, as it's there for a reason (defaulting respawn to this.manager.respawn)... can you test it out?

Reply:
i did actually test this, binding the first parameter to undefined still seems to cause the defaulting behavoir
```js
const fn = ((param = true) => param).bind(null, undefined);
console.log(fn()); // true, not undefined
```